evalutils/stats.py

```diff
@@ -100,33 +109,41 @@ def distance_transform_edt_float32(  # noqa: C901
     dt_inplace = isinstance(distances, np.ndarray)
 
     # calculate the feature transform
-    input = np.atleast_1d(np.where(input, 1, 0).astype(np.int8))
+    input_data = np.atleast_1d(input != 0)
+    input_data_check = input_data.nbytes >= gc_byte_threshold
+
+    def garbage_collect():
+        nonlocal input_data_check
+        if input_data_check:
+            gc.collect()
 
-    garbage_collect = gc.collect if input.nbytes > 100e6 else lambda: None
     garbage_collect()
 
-    input = input.astype(np.int32)
+    input_data = input_data.astype(np.int32)
     garbage_collect()
 
+    sampling_arr: Optional[np.ndarray] = None
```

Comment:
You're changing the side effects of the function here, right? `ndarrays` are passed by reference right, so in the original function this transform will affect `sampling` in the outer scope. 

This function is here to do the _same_ thing as it does in scipy, but only with lower memory. I don't understand why you have made this changes with no issue, and lumped in with some typing changes.

Reply:
Ah, I was not aware that it should affect the sampling array in the outer scope as well.
I did these changes to remove MyPy errors. I'll have a look at fixing both.

Reply:
Ok, this is tricky, first of all, sampling is not passed in as a `ndarray` and it will not be changed in the outer scope as a side-effect. The `_ni_support._normalize_sequence` method will convert it to a list if it isn't already one and the np.asarray will convert it to a `ndarray`, hence the original reference is no longer relevant and otherwise no significant changes will be made.

However, I agree that this might be a bit too much for this PR, so as discussed I'll create a different one.

Reply:
Ok, I reverted the method body to its old form. MyPy errors are back, but this PR is purer in form.